### PR TITLE
Missile Launcher

### PR DESCRIPTION
**Pull request by Rxann**
Thanks @Yimura  and qwlfjdekngh for the help

### CODE REVIEW

File: src/services/gui/gui_service.hpp [hunk 1]
```diff
@@ -160,26 +162,27 @@ namespace big
...
+void set_nav_size(int);
+void increment_nav_size();
+void reset_nav_size();
+std::map<tabs, navigation_struct>& get_navigation();
+void remove_from_nav(tabs existing_tab_id);
+}
+;
 
-	inline gui_service* g_gui_service{};
+inline gui_service* g_gui_service{};
```
Comment (Yimura) [inline src/services/gui/gui_service.hpp]: Indents gone

File: src/views/world/view_missile.cpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+#include "views/view.hpp"
+
+
+namespace big
+{
+	void view::missile()
+	{
+		ImGui::Text("Missle Launch");
+
+		static player_ptr victim = g_player_service->get_selected();
```
Comment (Yimura) [inline src/views/world/view_missile.cpp:14]: What happens if `get_selected()` is a nullptr?

**Review by Yimura — CHANGES_REQUESTED** [review]

File: src/views/world/view_missile.cpp [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+#include "pointers.hpp"
+#include "services/players/player_service.hpp"
+#include "views/view.hpp"
+
+
+namespace big
+{
+	void view::missile()
+	{
+		ImGui::Text("Missle Launch");
```
Comment (maybegreat48) [inline src/views/world/view_missile.cpp]: Typo

File: src/views/world/view_missile.cpp [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+					if (g.player.spectating)
+						g_player_service->set_selected(victim);
+				}
+			}
+			ImGui::EndCombo();
+		}
+		if (ImGui::Button("Launch"))
+		{
+			auto target = g_pointers->m_gta.m_ptr_to_handle(victim->get_ped());
+			auto firer  = self::ped;
```
Comment (maybegreat48) [inline src/views/world/view_missile.cpp]: Unused variables

File: docs/lua/tabs.md [hunk 5]
```diff
@@ -13,46 +13,88 @@ end)
...
+
 ### `GUI_TAB_SQUAD_SPAWNER`
+
 ### `GUI_TAB_CREATOR`
+
 ### `GUI_TAB_TRAIN`
+
 ### `GUI_TAB_BLACKHOLE`
+
+## `GUI_TAB_MISSILE`
```
Comment (gir489returns) [inline docs/lua/tabs.md:62]: Inconsistent with the others and you added a ton of spaces for no reason.

File: src/views/world/view_missile.cpp [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+			auto self = g_player_service->get_self();
+			if (ImGui::Selectable(self->get_name(), self->id() == victim->id()))
+				victim = self;
+			for (auto p : g_player_service->players() | std::ranges::views::values)
+			{
+				if (ImGui::Selectable(p->get_name(), p->id() == victim->id()))
+				{
+					victim = p;
+					if (g.player.spectating)
+						g_player_service->set_selected(victim);
```
Comment (gir489returns) [inline src/views/world/view_missile.cpp:28]: Inconsistency between using brackets and not using brackets. Pick one.

File: src/views/world/view_missile.cpp [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+		if (ImGui::BeginCombo("##victim", victim->get_name()))
+		{
+			auto self = g_player_service->get_self();
+			if (ImGui::Selectable(self->get_name(), self->id() == victim->id()))
+				victim = self;
+			for (auto p : g_player_service->players() | std::ranges::views::values)
+			{
+				if (ImGui::Selectable(p->get_name(), p->id() == victim->id()))
+				{
+					victim = p;
```
Comment (gir489returns) [inline src/views/world/view_missile.cpp:26]: What does p represent here? Useless variable name.

File: src/views/world/view_missile.cpp [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+					if (g.player.spectating)
+						g_player_service->set_selected(victim);
+				}
+			}
+			ImGui::EndCombo();
+		}
+		if (ImGui::Button("Launch"))
+		{
+			auto target = g_pointers->m_gta.m_ptr_to_handle(victim->get_ped());
```
Comment (gir489returns) [inline src/views/world/view_missile.cpp:35]: What is the purpose of this?

**Review by gir489returns — CHANGES_REQUESTED** [review]

File: src/views/world/view_missile.cpp (hunk 7 above)
Comment (Rxann) [reply src/views/world/view_missile.cpp]: It was taken from squad_spawner 🤣 ill fix that

**Comment by Yimura** [comment]
I'll be closing this in favour of this being transformed into a Lua script.

**Pull request closed without merging**